File: 1_algorithmic_toolbox/week2/python/2_last_digit_fibonacci.py
def fibonacci(n):
    if n <= 1:
        return n

    a, b = 0, 1
    for _ in range(n - 1):
        a, b = b, (a + b) % 10
    return b


if __name__ == "__main__":
    print(fibonacci(int(input())))

File: 1_algorithmic_toolbox/week2/rust/2_last_digit_fibonacci.rs
fn fibonacci(n: i32) -> i32 {
    if n <= 1 {
        return n;
    }

    let (mut a, mut b) = (1, 0);
    for _ in 0..n - 1 {
        a = a + b;
        b = a - b;
        a %= 10;
    }
    a
}

fn main() {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    let n = input.trim().parse::<i32>().unwrap();
    println!("{}", fibonacci(n));
}
